Style/Documentation recognizes 'Const = Class.new' as a 'class definition'

alexdowad opened the pull request:
...and likewise for modules.

bbatsov commented on lib/rubocop/cop/style/documentation.rb:
```diff
@@ -47,18 +47,25 @@ def check(ast, ast_with_comments)
+          node.class_type? ||
+            node.module_type? ||
+            (node.casgn_type? &&
+              (rhs = node.children[2]).send_type? &&
```
I don't like the usage of array references and `children` in general, as it really obscures what you're trying to do. Even I'm not sure about it, so people not familiar with `parser` will be extra confused. Individual node destructuring is more verbose, but also way easier to comprehend as can give sensible names to everything you care about. I'd also say that the entire boolean expression is pretty hard to comprehend/debug, so I'd split it into smaller chunks. 

It's also a bit debatable whether a class check is a good idea in something named `module_definition?`.

alexdowad replied:
I'll try to find a way to make the code read better. Regarding `module_definition?`, I chose that name because (technically) Ruby classes are also modules, though many programmers may think of them as distinct.

bbatsov replied:
Yeah, I'm well aware of this, but as you said - others might not be. :-) That's not a major point, though.

alexdowad replied:
I'm trying to come up with a way to make complex matching operations more concise. I've created a `Pattern` class which does recursive pattern matching on AST nodes. Will send a PR soon so you can have a look.

alexdowad commented:
Just rebased this one `master` and fixed it up. I know @bbatsov had a concern about whether `#module_definition?` would be easily understandable or not -- hopefully the added comment will help?

bbatsov commented:
Apart from the failing build, the code looks OK now.

alexdowad commented:
@bbatsov Sorry about that, fixed.
